algebras/algebra/src/test/scala/endpoints4s/algebra/client/StubServer.scala

```diff
@@ -0,0 +1,437 @@
+package endpoints4s.algebra.client
+
+import java.net.ServerSocket
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model.HttpMethods._
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.model.headers._
+import akka.http.scaladsl.unmarshalling.Unmarshal
+import akka.stream.scaladsl.Sink
+import akka.stream.scaladsl.Source
+import akka.util.ByteString
+
+import scala.concurrent.ExecutionContextExecutor
+import scala.concurrent.Future
+import scala.util.Failure
+import scala.util.Success
+import scala.util.Try
+
+// can start with sbt "algebraJVM/test:runMain endpoints4s.algebra.client.StubServer 8080"
+// Win - Start-Process -NoNewWindow sbt -ArgumentList "`"algebraJVM/test:runMain endpoints4s.algebra.client.StubServer 8080`""
+// Linux - sbt "algebraJVM/test:runMain endpoints4s.algebra.client.StubServer 8080" &
+// Use sbt-revolver in development - project algebraJVM; set test/reStartArgs := Seq("8080"); set test/reStart/mainClass := Some("endpoints4s.algebra.client.StubServer"); reStart.
+// sbt-revolver not work because of https://github.com/spray/sbt-revolver/issues/82.
+object StubServer extends App {
```

Comment:
Could not decide if this should be moved to a separate SBT project. It does make sense to separate `StubServer` lifecycle and dependencies from the actual project, and avoiding the https://github.com/spray/sbt-revolver/issues/82 issue with `sbt-revolver` would also allow its neat use during development.

Reply:
It seems fine to make it a separate project. What would be the drawbacks of doing so?

Reply:
Cannot think of any, only thing that comes to mind is that because repository's root is an SBT project it feels kind of funky to have a SBT project within a SBT project :) I will separate it out.

Reply:
I am replying to https://github.com/endpoints4s/endpoints4s/pull/959#issuecomment-974909411 here to continue the thread.

So do you think it should be a sub-module of endpoints4s sbt project or a separate sbt project?

Reply:
A sub-module. Sorry for not being clear!